src/query/service/src/interpreters/interpreter_execute_immediate.rs

```diff
@@ -0,0 +1,425 @@
+
+#[derive(Debug)]
+pub struct ExecuteImmediateInterpreter {
+    ctx: Arc<QueryContext>,
+    plan: ExecuteImmediatePlan,
+}
+
+impl ExecuteImmediateInterpreter {
+    pub fn try_create(ctx: Arc<QueryContext>, plan: ExecuteImmediatePlan) -> Result<Self> {
+        Ok(ExecuteImmediateInterpreter { ctx, plan })
+    }
+}
+
+#[async_trait::async_trait]
+impl Interpreter for ExecuteImmediateInterpreter {
+    fn name(&self) -> &str {
+        "ExecuteImmediateInterpreter"
+    }
+
+    fn is_ddl(&self) -> bool {
+        false
+    }
+
+    #[minitrace::trace]
+    #[async_backtrace::framed]
+    async fn execute2(&self) -> Result<PipelineBuildResult> {
+        let res: Result<_> = try {
+            let dialect = self.ctx.get_settings().get_sql_dialect()?;
+            let tokens = tokenize_sql(&self.plan.script)?;
+            let mut ast = run_parser(&tokens, dialect, ParseMode::Template, false, script_block)?;
+
+            let mut src = vec![];
+            for declare in ast.declares {
+                match declare {
+                    DeclareItem::Var(declare) => src.push(ScriptStatement::LetVar { declare }),
+                    DeclareItem::Set(declare) => {
+                        src.push(ScriptStatement::LetStatement { declare })
+                    }
+                }
+            }
+            src.append(&mut ast.body);
+            let compiled = compile(&src)?;
+
+            let client = ScriptClient {
+                ctx: self.ctx.clone(),
+            };
+            let mut executor = Executor::load(ast.span, client, compiled);
+            let result = executor.run(/* TODO(andylokandy) */ 1000).await?;
+
+            match result {
+                Some(ReturnValue::Var(scalar)) => {
+                    PipelineBuildResult::from_blocks(vec![DataBlock::new_from_columns(vec![
+                        StringType::from_data(vec![scalar.to_string()]),
+                    ])])?
+                }
+                Some(ReturnValue::Set(set)) => {
+                    let rendered_table = box_render(
+                        &set.schema,
+                        &[set.block.clone()],
+                        usize::MAX,
+                        usize::MAX,
+                        usize::MAX,
+                        true,
+                    )?;
+                    let lines = rendered_table.lines().map(|x| x.to_string()).collect();
+                    PipelineBuildResult::from_blocks(vec![DataBlock::new_from_columns(vec![
+                        StringType::from_data(lines),
+                    ])])?
+                }
+                None => PipelineBuildResult::from_blocks(vec![DataBlock::new_from_columns(vec![
+                    StringType::from_data(Vec::<String>::new()),
+                ])])?,
+            }
+        };
+
+        res.map_err(|err| err.display_with_sql(&self.plan.script))
+    }
+}
+
+#[derive(Debug, Clone)]
+struct QueryResult {
+    schema: DataSchemaRef,
+    block: DataBlock,
```

Comment:
why not use `blocks:  Vec<DataBlock>`

Reply:
if so, the iter will be more complicated. In most cases, we will only get one block returned, right?

Reply:
It's right.  We can improve it later.